jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -801,7 +812,14 @@ public Response payAllInvoices(@PathParam("accountId") final UUID accountId,
             final BigDecimal amountToPay = (remainingRequestPayment.compareTo(invoice.getBalance()) >= 0) ?
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
-                final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
+                UUID paymentMethodId;
+                if(inputPaymentMethodId == null){
+                    paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
+                }
+                else{
+                    paymentMethodId = inputPaymentMethodId;
+                }
```

Comment:
Nit: Same formatting remarks as above.

Reply:
Also, by default if `externalPayment` is not set, the code will assume a default of false; therefore, it would make sense to assume that if this is true, this was intentional, and you could rewrite it:
```
final UUID paymentMethodId = externalPayment ? account.getPaymentMethodId() : inputPaymentMethodId;
```

Reply:
Ok, Thanks for the tips.